src/goto-symex/goto_symex.cpp

```diff
@@ -665,3 +681,181 @@ bool goto_symext::constant_propagate_delete_char_at(
 
   return true;
 }
+
+bool goto_symext::constant_propagate_case_change(
+  statet &state,
+  symex_assignt &symex_assign,
+  const function_application_exprt &f_l1,
+  bool to_upper)
+{
+  const auto &f_type = to_mathematical_function_type(f_l1.function().type());
+  const auto &length_type = f_type.domain().at(0);
+  const auto &char_type = to_pointer_type(f_type.domain().at(1)).subtype();
+
+  const refined_string_exprt &s = to_string_expr(f_l1.arguments().at(2));
+  const auto s_data_opt = try_evaluate_constant_string(state, s.content());
+
+  if(!s_data_opt)
+    return false;
+
+  array_exprt string_data = s_data_opt->get();
+
+  auto &operands = string_data.operands();
+  for(auto &operand : operands)
+  {
+    auto &constant_value = to_constant_expr(operand);
+    char character = numeric_cast_v<int>(constant_value);
+    if(isalpha(character))
+    {
+      if(to_upper)
+      {
+        if(islower(character))
+        {
+          std::stringstream stream;
+          stream << std::hex << toupper(character);
+          constant_value.set_value(stream.str());
```

Comment:
Honestly these are 16-bit ints as far as CBMC is concerned, I'd use `from_integer` rather than hope you get the expected formatting for `constant_exprt`'s value member

Reply:
Like this for example: https://github.com/diffblue/cbmc/blob/73e410ab653edffe27e4a74db86169d0a627443f/src/solvers/strings/string_constraint_generator_code_points.cpp#L40